src/liballoc_system/lib.rs

```diff
@@ -39,29 +39,35 @@ const MIN_ALIGN: usize = 8;
 const MIN_ALIGN: usize = 16;
 
 #[no_mangle]
-pub extern fn __rust_allocate(size: usize, align: usize) -> *mut u8 {
+pub extern "C" fn __rust_allocate(size: usize, align: usize) -> *mut u8 {
```

Comment:
I was a little surprised too, but I guess we should prefer one or the other. Is bare extern preferred?

(Not blocking on this since we can change rustfmt to change them back).